### PR TITLE
Database implementation for the Wrapper

### PR DESCRIPTION
**Pull request by McRuben**
- [ ] breaking changes
- [x] no breaking changes

### changes made to the repository
added some methods to access the nodes database

### CODE REVIEW

File: cloudnet-wrapper-jvm/src/main/java/de/dytanic/cloudnet/wrapper/database/IDatabase.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    Collection<JsonDocument> documents();
+
+    Map<String, JsonDocument> entries();
+
+    void iterate(BiConsumer<String, JsonDocument> consumer);
+
+    void clear();
+
+    /*= ----------------------------------------------------------- =*/
```
Comment (realPanamo) [inline cloudnet-wrapper-jvm/src/main/java/de/dytanic/cloudnet/wrapper/database/IDatabase.java]: Please remove stupid comments like this

File: cloudnet/src/main/java/de/dytanic/cloudnet/network/listener/PacketServerDatabaseAction.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+import de.dytanic.cloudnet.database.IDatabase;
+import de.dytanic.cloudnet.driver.network.INetworkChannel;
+import de.dytanic.cloudnet.driver.network.def.PacketConstants;
+import de.dytanic.cloudnet.driver.network.protocol.IPacket;
+import de.dytanic.cloudnet.driver.network.protocol.IPacketListener;
+import de.dytanic.cloudnet.driver.network.protocol.Packet;
+
+import java.util.UUID;
+
+public class PacketServerDatabaseAction implements IPacketListener {
```
Comment (realPanamo) [inline cloudnet/src/main/java/de/dytanic/cloudnet/network/listener/PacketServerDatabaseAction.java]: Leave a blank line after the class declaration

File: cloudnet/src/main/java/de/dytanic/cloudnet/network/listener/PacketServerDatabaseAction.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        sendResponse(channel, uniqueId, header, null);
+    }
+
+    private void sendResponse(INetworkChannel channel, UUID uniqueId, JsonDocument header, byte[] body) {
+        channel.sendPacket(new Packet(PacketConstants.INTERNAL_CALLABLE_CHANNEL, uniqueId, header, body));
+    }
+
+    private void sendEmptyResponse(INetworkChannel channel, UUID uniqueId) {
+        sendResponse(channel, uniqueId, new JsonDocument(), null);
+    }
```
Comment (realPanamo) [inline cloudnet/src/main/java/de/dytanic/cloudnet/network/listener/PacketServerDatabaseAction.java]: Blank line would be nice after this too

**Comment by realPanamo** [comment]
Only a few style issues, but great job in general.

File: cloudnet/src/main/java/de/dytanic/cloudnet/network/listener/PacketServerDatabaseAction.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    @Override
+    public void handle(INetworkChannel channel, IPacket packet) throws Exception {
+        if (packet.getHeader().contains("message")) {
+            AbstractDatabaseProvider databaseProvider = CloudNet.getInstance().getDatabaseProvider();
+            String message = packet.getHeader().getString("message");
+
+            if (packet.getHeader().contains("database")) {
+                String databaseName = packet.getHeader().getString("database");
+
+                IDatabase database = databaseProvider.getDatabase(databaseName);
```
Comment (realPanamo) [inline cloudnet/src/main/java/de/dytanic/cloudnet/network/listener/PacketServerDatabaseAction.java]: You might want to check if the database is null

Comment (McRuben) [reply cloudnet/src/main/java/de/dytanic/cloudnet/network/listener/PacketServerDatabaseAction.java]: > You might want to check if the database is null

The database should never be null

Comment (realPanamo) [reply cloudnet/src/main/java/de/dytanic/cloudnet/network/listener/PacketServerDatabaseAction.java]: Ah, yes, you're right

**Pull request merged**